internal/filedetector/internal/inotifyinfo/inotifyinfo.go

```diff
@@ -0,0 +1,111 @@
+package inotifyinfo
+
+import (
+	"encoding/json"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/alloy/internal/runtime/logging/level"
+	"github.com/prometheus/procfs"
+)
+
+type inotifyInfo struct {
+	ProcInotifyInfos []procInotifyInfo `json:"proc_inotify_infos"`
+	MaxQueuedEvents  int               `json:"max_queued_events"`
+	MaxUserInstances int               `json:"max_user_instances"`
+	MaxUserWatches   int               `json:"max_user_watches"`
+}
+
+type procInotifyInfo struct {
+	PID              int    `json:"pid"`
+	ProcessName      string `json:"process_name"`
+	InotifyWatchers  int    `json:"inotify_watchers"`
+	InotifyInstances int    `json:"inotify_instances"`
+}
+
+func DiagnosticsJson(logger log.Logger) string {
+	return diagnostics(logger).json(logger)
```

Comment:
This is destined to panic in case of error in `diagnostics`.

Reply:
Actually it doesn't panic if diagnostics returns a nil. But I'll add some tests, I agree it's important to make it clear.